frontend/src/LandingPage.js

```diff
@@ -1,50 +1,269 @@
+            markers.splice(i,1);
+            break;
+          }
+        }
+
+        // Update the markers on map page.
+        for (let i = 0; i < markers.length; i++) {
+          markers[i].markerObject.setMap(map);
+        }
+    }
+    function addMarker(location) {
+        // Convert string location to JSON object.   
+        const position = {
+            lat: parseFloat(location.split(',')[0]),
+            lng: parseFloat(location.split(',')[1]),
+        };
+        map.setCenter(position);
+
+        const markerObject = new google.maps.Marker({
+          position: position,
+        });
+
+        const marker = {
+          position: position,
+          markerObject: markerObject,
+        }
+
+        // Add new marker to the array.
+        markers.push(marker);
+
+        // Update markers on map page.
+        for (let i = 0; i < markers.length; i++) {
+          markers[i].markerObject.setMap(map);
+        }
+    }
+    let lat1, lat2, long1, long2;
+    const [hiddenIn, setHiddenIn] = useState(false);
+    const [hiddenOut, setHiddenOut] = useState(true);
+    function authentication() {
+        
+        fetch('/api/login')
+            .then((response) => response.json())
+            .then((user) => {
+                const loginSection = document.getElementById('loginsection');
+                const loginMessage = document.getElementById('login');
+                const logoutSection = document.getElementById('logoutsection');
+                const logoutMessage = document.getElementById('logout');
+                
+                if (user.loggedIn) {
+                    loginSection.style.display = 'none';
+                    logoutMessage.href = user.url;
+                    logoutSection.style.display = 'block';
+                    
+                } else {
+                    logoutSection.style.display = 'none';
+                    loginMessage.href = user.url;
+                    loginSection.style.display = 'block';
+                    
+                }
+                currentUserId = user.id;
+                localStorage.setItem("id", currentUserId);
+            });
+    }
+
+    function calls() {
+        handleScriptLoad1(setQuery1);
+        handleScriptLoad2(setQuery2);
+    }
+
+    const loadScript = (url) => {
+        let script = document.createElement("script");
+        script.type = "text/javascript";
+
+        if (script.readyState) {
+            script.onreadystatechange = function() {
+                if (script.readyState === "loaded" || script.readyState === "complete") {
+                    script.onreadystatechange = null;
+                    calls();
+                }
+            };
+        } else {
+            script.onload = () => calls();
+        }
+
+        script.src = url;
+        document.getElementsByTagName("head")[0].appendChild(script);
+    };
+    function handleScriptLoad1(updateQuery) {
```

Comment:
looks very similar to the next function. Consider combining it (with an element id as a parameter)

Reply:
Oh. I just noticed that they use different class variables autoComplete1 and autoComplete2. Again, it's better to rename the forms. If you can come up with a good way to merge the code, please go ahead

Reply:
Fixed, thanks.